youtube_dl/extractor/gbnews.py

```diff
@@ -0,0 +1,152 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..compat import (
+    compat_str,
+)
+from ..utils import (
+    extract_attributes,
+    ExtractorError,
+    try_get,
+)
+
+
+class GBNewsIE(InfoExtractor):
+    '''GB News clips and features'''
+
+    # \w+ is normally shows or news, but apparently any word redirects to the correct URL
+    _VALID_URL = r'https?://(?:www\.)?gbnews\.uk/(?:\w+(?:/(?P<display_id>[^/]+))?|a)/(?P<id>\d+)'
+    _PLATFORM = 'safari'
+    _SSMP_URL = 'https://mm-dev.simplestream.com/ssmp/api.php'
+    _TESTS = [{
+        'url': 'https://www.gbnews.uk/shows/andrew-neils-message-to-companies-choosing-to-boycott-gb-news/106889',
+        'info_dict': {
+            'id': '106889',
+            'ext': 'mp4',
+            'title': "Andrew Neil's message to companies choosing to boycott GB News",
+            'description': 'md5:b281f5d22fd6d5eda64a4e3ba771b351',
+        },
+    },
+    ]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
```

Comment:
Support deeper URL paths:
```suggestion
        display_id = self._match_id(url).split('/')[-1]
```